### ISSUE TITLE
Some songs not playing through/are hanging (missing chunks?)

### ISSUE THREAD

**Issue by herrernst** [plietar/librespot#14]
Some songs (I don't know if specific ones or if happens at random) get stuck playing beneath the end. From the output, I see that the first and last (why?) chunks are loaded first, and the track is marked loaded. Then other chunks are loaded. The last one is 58 (but 59 is missing!), then nothing happens, and when playback reaches the last ~20 seconds of the song, playback is stuck and no control is possible anymore. It happend to me with the track `spotify:track:5ERPZNZfMPF5j0xgEnoQoV`:

```
Chunk 0
Chunk 1
Chunk 2
Chunk 61
Chunk 3
Chunk 60
Chunk 4
Load Done
Chunk 5
Chunk 6
Chunk 7
Chunk 8
Chunk 9
Chunk 10
Chunk 11
Chunk 12
Chunk 13
Chunk 14
Chunk 15
Chunk 16
Chunk 17
Chunk 18
Chunk 19
Chunk 20
Chunk 21
Chunk 22
Chunk 23
Chunk 24
Chunk 25
Chunk 26
Chunk 27
Chunk 28
Chunk 29
Chunk 30
Chunk 31
Chunk 32
Chunk 33
Chunk 34
Chunk 35
Chunk 36
Chunk 37
Chunk 38
Chunk 39
Chunk 40
Chunk 41
Chunk 42
Chunk 43
Chunk 44
Chunk 45
Chunk 46
Chunk 47
Chunk 48
Chunk 49
Chunk 50
Chunk 51
Chunk 52
Chunk 53
Chunk 54
Chunk 55
Chunk 56
Chunk 57
Chunk 58
^C
```

(Chunk 60 and 61 are loaded at the beginning, 59 is missing)

If it matters, I build with latest Rust nightly (b4707ebca 2015-12-27) on Linux x64 with one maybe troublesome warning:

```
src/player.rs:219:9: 219:22 warning: unreachable statement, #[warn(unreachable_code)] on by default
src/player.rs:219         drop(stream);
                          ^~~~~~~~~~~~~
```

Thank you!

**Comment by herrernst** [comment]
Could it be the `/ 2` in https://github.com/plietar/librespot/blob/master/src/audio_file.rs#L56 ? It wouldn't fetch all chunks if the last chunk would be less than half the chunk size, would it? I changed it to `(size + CHUNK_SIZE) / CHUNK_SIZE;`, and it seems to work now.

**Comment by plietar** [comment]
I did notice something similar the other day, but I didn't have time to investigate.

Your explanation makes sense, although it should probably be `(size + CHUNK_SIZE - 1) / CHUNK_SIZE` to take into account the case where `size` is a multiple of `CHUNK_SIZE`.
Thanks for taking the time to look into it. Do you mind making a PR with it please ?

Regarding the chunks 60 & 61 loaded at the beginning, this is expected behavior. libvorbis, which librespot uses for decoding, seeks to the end of the file when it loads the file. It probably does it simply to know the size of the file, but I haven't looked into it more than that.
EDIT: for reference, this is where libvorbis seeks to the end :
https://git.xiph.org/?p=vorbis.git;a=blob;f=lib/vorbisfile.c;h=8a0a2801559a6736a08dd5d2273f7f02f7dda2ec;hb=refs/heads/master#l627

**Comment by plietar** [comment]
Fixed by 525b27d

**Comment by herrernst** [comment]
You're right of course, thank you very much!
